### PR TITLE
[docs] add new FAQ entries

### PR DESCRIPTION
**Pull request by iomintz**
## Summary

- How to DM
- Lack of search and its implications
  - This one raises a concern, namely that most people asking about the channel search API are asking bout overall message counts. Instead of not suggesting `Messageable.history()`, I opted to answer the question at face value and include a warning for those looking for this particular use case. In the warning, should I mention that EUD should be encrypted?
- Where to find examples
- Getting the ID of a sent message
  - Should I include a specific example of something to do with the ID? I couldn't think of anything…

(There is also an Easter egg in this PR, can you find it?)

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: docs/faq.rst [hunk 1]
```diff
@@ -76,6 +76,28 @@ General
 
 General questions regarding library usage belong here.
 
+Are there any usage examples?
```
Comment (Rapptz) [inline docs/faq.rst]:
```suggestion
Where can I find examples?
```

File: docs/faq.rst [hunk 2]
```diff
@@ -76,6 +76,28 @@ General
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Example code can be found in the `examples/ <https://github.com/Rapptz/discord.py/tree/master/examples>`_ folder
+in the repository.
+
+Can I search messages in channels?
```
Comment (Rapptz) [inline docs/faq.rst]: I don't like this question's title nor its content. It needs a rewrite.

File: docs/faq.rst [hunk 3]
```diff
@@ -104,8 +126,27 @@ You must fetch the channel directly and then call the appropriate method. Exampl
     channel = client.get_channel(12324234183172)
     await channel.send('hello')
 
+How do I send a DM?
+~~~~~~~~~~~~~~~~~~~
+
+Get the :class:`User` object and call :meth:`User.send`. For example: ::
```
Comment (Rapptz) [inline docs/faq.rst]:
```suggestion
Get the :class:`User` or :class:`Member` object and call :meth:`abc.Messageable.send` on it. For example: ::
```

File: docs/faq.rst [hunk 4]
```diff
@@ -104,8 +126,27 @@ You must fetch the channel directly and then call the appropriate method. Exampl
...
+
+	user = client.get_user(381870129706958858)
+	await user.send('👀')
```
Comment (Rapptz) [inline docs/faq.rst]: Also needs an example showing `message.author.send` since that's 99.99999% the more common use case.

File: docs/faq.rst [hunk 5]
```diff
@@ -104,8 +126,27 @@ You must fetch the channel directly and then call the appropriate method. Exampl
...
+
+If the user is not cached (:meth:`Client.get_user` returns ``None``), you can use :meth:`Client.fetch_user` instead.
```
Comment (Rapptz) [inline docs/faq.rst]: This entire sentence needs to be removed. If it's not in the cache then the DM will just fail.
```suggestion
```

File: docs/faq.rst [hunk 6]
```diff
@@ -104,8 +126,27 @@ You must fetch the channel directly and then call the appropriate method. Exampl
...
+
+If you are responding to an event, such as :func:`on_message`, you already have the :class:`User` object via :attr:`Message.author`..
```
Comment (Rapptz) [inline docs/faq.rst]:
```suggestion
```

This sentence is better off being shown in the code sample rather than written text.

File: docs/faq.rst [hunk 7]
```diff
@@ -104,8 +126,27 @@ You must fetch the channel directly and then call the appropriate method. Exampl
...
+Use :meth:`User.send`.
+
+How do I get the ID of a sent message?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:meth:`abc.Messageable.send` returns the :class:`Message` that was sent.
+The ID of a message can be accessed via :attr:`Message.id`.
```
Comment (Rapptz) [inline docs/faq.rst]: As silly as this is, this needs a code example again. Remember the target audience.

File: docs/faq.rst (hunk 2 above)
Comment (iomintz) [reply docs/faq.rst]: I ended up deleting that entry because I'm not sure how best to go about it.

**Pull request closed without merging**